flocker/provision/_install.py

```diff
@@ -316,48 +315,85 @@ def install_cli_commands_ubuntu(distribution, package_source):
+    # Install package and all dependencies
+    commands.append(run_from_args([
+        'apt-get', '-y', '--force-yes', 'install', package_name]))
 
     return sequence(commands)
 
 
 _task_install_commands = {
-    'centos-7': install_cli_commands_yum,
-    'ubuntu-14.04': install_cli_commands_ubuntu,
-    'ubuntu-15.04': install_cli_commands_ubuntu,
+    'centos-7': install_commands_yum,
+    'ubuntu-14.04': install_commands_ubuntu,
+    'ubuntu-15.04': install_commands_ubuntu,
 }
 
 
-def task_cli_pkg_install(distribution, package_source=PackageSource()):
+def task_package_install(package_name, distribution,
+                         package_source=PackageSource()):
     """
-    Install flocker CLI on a distribution.
+    Install Flocker package on a distribution.
 
     The ClusterHQ repo is added for downloading latest releases.  If
     ``package_source`` contains a branch, then a BuildBot repo will also
     be added to the package search path, to use in-development packages.
     Note, the ClusterHQ repo is always enabled, to provide dependencies.
 
+    :param str package_name: The name of the package to install.
+    :param bytes distribution: The distribution the node is running.
+    :param PackageSource package_source: The source from which to install the
+        package.
+
+    :return: a sequence of commands to run on the distribution
+    """
+    if package_source.branch:
+        # A development branch has been selected - add its Buildbot repo
+        result_path = posixpath.join(
+            '/results/omnibus/', package_source.branch, distribution)
+        base_url = urljoin(package_source.build_server, result_path)
+    else:
+        base_url = None
+
+    try:
+        installer = _task_install_commands[distribution]
+    except KeyError:
+        raise UnsupportedDistribution()
+    return installer(package_name, distribution, package_source,
+                     base_url)
+
+
+def task_cli_pkg_install(distribution, package_source=PackageSource()):
+    """
+    Install the Flocker CLI package.
+
     :param bytes distribution: The distribution the node is running.
     :param PackageSource package_source: The source from which to install the
         package.
 
     :return: a sequence of commands to run on the distribution
     """
-    return _task_install_commands[distribution](distribution, package_source)
+    commands = task_package_install("clusterhq-flocker-cli", distribution,
+                                    package_source)
+    if is_ubuntu(distribution):
+        # We want to use sudo for better documentation output, even though
+        # it's not actually necessary since this will run as root.
+        return sequence([
+            (Effect(Sudo(command=e.intent.command,
+                         log_command_filter=e.intent.log_command_filter))
+             if isinstance(e.intent, Run) else e)
+            for e in commands.intent.effects])
```

Comment:
This is pretty ugly and slightly obscure. Maybe pass Run/Sudo as a parameter to `task_package_install` :question: